src/v/kafka/server/group.cc

```diff
@@ -2203,13 +2203,16 @@ group::store_txn_offsets(txn_offset_commit_request r) {
 
     prepared_tx ptx;
     ptx.tx_seq = tx_seq;
+    const auto now = model::timestamp::now();
     for (const auto& [tp, offset] : offsets) {
         offset_metadata md{
-          .log_offset = e.value().last_offset,
-          .offset = offset.offset,
-          .metadata = offset.metadata.value_or(""),
-          .committed_leader_epoch = kafka::leader_epoch(offset.leader_epoch)};
-        ptx.offsets[tp] = md;
+          e.value().last_offset,
```

Comment:
Can we keep `.property = ...` syntax? With it we don't need to jump to definitions to check if the order is correct so it's easier to review and harder to introduce an error

Reply:
sure, but it's a trade-off. to solve the ordering issue you allude to strong types can be used. but, maintaining aggregate initialization support has the downside that there is no straight forward way to know if i have updated all the call sites.

i'm happy to drop those changes in a PR revision because by adding the constructor i was able to discover all of the call sites that exited.

for the review, i've separated out that change, so if you look at the individual commits they don't have all that noise.

Reply:
fixed